### PR TITLE
Content Provider analytics

### LINKED ISSUE #4576: Add analytics to Content Providers
Blocked by #4462 

We want to be able to find out if any apps are using our Content Providers for forms or instances.

## Acceptance

- [x] Given there is a call to either `insert`, `update` or `delete` on the `FormsProvider`
When I navigate to the analytics dashboard
Then I see there has been an event named `"FormsProvider#<method>`

- [x] Given there is a call to either `insert`, `update` or `delete` on the `InstanceProvider`
When I navigate to the analytics dashboard
Then I see there has been an event named `"InsanceProvider#<method>`

- [x] Given there is a call to `FormsProvider#query` from outside Collect
When I navigate to the analytics dashboard
Then I see there has been an event named `"FormsProvider#query`

- [x] Given there is a call to `InstanceProvider#query` from outside Collect
When I navigate to the analytics dashboard
Then I see there has been an event named `"InstanceProvider#query`

## Note

The URIs we use will be different for internal and external apps as internal URIs will need to include a project in some way whereas external ones will not (they use old URIs and the "first" project is assumed to be being referred to). This will let us differentiate between internal (from `CursorLoader`s etc) and external calls.

Every event should include server hash as well (using `Analytics#logServerEvent`).

### PR DESCRIPTION
**Pull request by seadowg**
Closes #4576

I also reworked the analytics interface a little so it doesn't include anything around servers and forms. Logic around that go moved to `AnalyticsUtils`.

#### What has been done to verify that this works as intended?

Ran tests.

#### Why is this the best possible solution? Were any other approaches considered?

Only thing of note here is that I needed a way to differentiate between internal and external query calls. To do this, I made `CursorLoaderFactory` add a query param to the URI it sends so the content provider can identify calls that it's responsible for.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Shouldn't be any risk here really, as it's just adding analytics. 

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/external/FormsProvider.java [hunk 1]
```diff
@@ -108,6 +115,11 @@ public Cursor query(@NonNull Uri uri, String[] projection, String selection, Str
 
         String projectId = getProjectId(uri);
 
+        // We only want to log external calls to the content provider
+        if (uri.getQueryParameter(CursorLoaderFactory.INTERNAL_QUERY_PARAM) != null) {
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/external/FormsProvider.java]: If you want to log external calls it should be == null right?

File: collect_app/src/main/java/org/odk/collect/android/external/InstanceProvider.java [hunk 2]
```diff
@@ -82,6 +89,11 @@ public Cursor query(@NonNull Uri uri, String[] projection, String selection, Str
 
         String projectId = getProjectId(uri);
 
+        // We only want to log external calls to the content provider
+        if (uri.getQueryParameter(CursorLoaderFactory.INTERNAL_QUERY_PARAM) != null) {
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/external/InstanceProvider.java]: The same as above, should be == null.

**Review by grzesiek2010 — CHANGES_REQUESTED** [review]

File: collect_app/src/main/java/org/odk/collect/android/external/FormsProvider.java (hunk 1 above)
Comment (seadowg) [reply collect_app/src/main/java/org/odk/collect/android/external/FormsProvider.java]: 😔 yes. That's a silly mistake!

File: collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java [hunk 3]
```diff
@@ -174,36 +172,26 @@ public void openHierarchy() {
...
+        AnalyticsUtils.logFormEvent(event);
     }
 
     public NonNullLiveData<Boolean> hasBackgroundRecording() {
         return hasBackgroundRecording;
     }
 
-    private String getFormIdentifierHash() {
-        if (formController != null) {
-            return formController.getCurrentFormIdentifierHash();
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java:186]: I think this method in `FormController` is no longer used.

**Review by grzesiek2010 — CHANGES_REQUESTED** [review]

File: collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java (hunk 3 above)
Comment (seadowg) [reply collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java:186]: Ah yeah good spot

**Review by grzesiek2010 — APPROVED** [review]
I think we should use the same way of logging events across the app because now sometimes it's `Analytics.log` (static methods) sometimes we use objects generated by dagger, but it's a thing for another pr.

**Pull request merged**